src/wasm2js.h

```diff
@@ -2283,25 +2304,35 @@ void Wasm2JSGlue::emitMemory(
 
   out << "var bufferView = new Uint8Array(" << buffer << ");\n";
 
-  for (auto& seg : wasm.memory.segments) {
-    assert(!seg.isPassive && "passive segments not implemented yet");
-    out << "base64DecodeToExistingUint8Array(bufferView, " << globalOffset(seg)
-        << ", \"" << base64Encode(seg.data) << "\");\n";
+  for (Index i = 0; i < wasm.memory.segments.size(); i++) {
+    auto& seg = wasm.memory.segments[i];
+    if (!seg.isPassive) {
+      // Plain active segments are decoded directly into the main memory.
+      out << "base64DecodeToExistingUint8Array(bufferView, "
+          << globalOffset(seg) << ", \"" << base64Encode(seg.data) << "\");\n";
+    } else {
+      // Fancy passive segments are decoded into typed arrays on the side, for
+      // later copying.
+      out << "memorySegments[" << i
+          << "] = base64DecodeToExistingUint8Array(new Uint8Array("
+          << seg.data.size() << ")"
+          << ", 0, \"" << base64Encode(seg.data) << "\");\n";
+    }
   }
 }
 
-void Wasm2JSGlue::emitScratchMemorySupport() {
+void Wasm2JSGlue::emitSpecialSupport() {
   // The scratch memory helpers are emitted here the glue. We may also want to
```

Comment:
This might be a good time to fix the first sentence of this comment. I'm not exactly sure what it's supposed to say.